src/main/java/ps/as/TransactionDao.java

```diff
@@ -0,0 +1,96 @@
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void addTransaction(Transaction transaction) {
+        String sql = "INSERT INTO transaction (type, description, amount, date) VALUES (?,?,?,?)";
+        PreparedStatement preparedStatement;
+        try {
+            preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, String.valueOf(transaction.getType()));
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setDouble(3, transaction.getAmount());
+            preparedStatement.setString(4, transaction.getDate());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+               transaction.setId(generatedKeys.getInt(1));
+           }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public boolean modifyTransaction(Transaction transaction) {
+        String sql = "UPDATE transaction SET type = ?, description = ?, amount = ?, date = ? WHERE id = ?";
+        PreparedStatement preparedStatement;
+        try {
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, String.valueOf(transaction.getType()));
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setDouble(3, transaction.getAmount());
+            preparedStatement.setString(4, transaction.getDate());
+            preparedStatement.setInt(5, transaction.getId());
+            int updatedRows = preparedStatement.executeUpdate();
+            return updatedRows != 0;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public boolean deleteTransaction(int id) {
+        String sql = "DELETE FROM transaction WHERE id = ?";
+        PreparedStatement preparedStatement;
+        try {
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, id);
+            int updatedRows = preparedStatement.executeUpdate();
+            return updatedRows != 0;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public ResultSet findAllIncome() {
+        String sql = "SELECT * FROM transaction WHERE type = 'P'";
+        PreparedStatement preparedStatement;
+        try {
+            preparedStatement = connection.prepareStatement(sql);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return resultSet;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public ResultSet findAllExpenses() {
```

Comment:
ResultSet to jest rzecz charakterystyczna dla baz danych. Powinnaś tutaj zwracać kolekcję, czyli List, lub Set z obiektami Transaction.